common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -461,7 +461,7 @@ object DotcomRenderingDataModel {
       blocks.exists(block => DotcomRenderingUtils.stringContainsAffiliateableLinks(block.bodyHtml))
     }
 
-    val shouldAddAffiliateLinks = DotcomRenderingUtils.shouldAddAffiliateLinks(content)
+    val shouldAddAffiliateLinks = DotcomRenderingUtils.shouldAddAffiliateLinks(content)(request)
```

Comment:
Tidy up for later: We shouldn't have to do this, if we made the request implicitly available on `forLiveBlog`'s `apply` function. 

Also isn't it weird that all this logic lives in `forLiveBlog` and not `forArticle`? Again, something to dig into later.